lib/streamaggr/total.go

```diff
@@ -23,13 +23,7 @@ type totalAggrState struct {
 	// Time series state is dropped if no new samples are received during stalenessSecs.
 	//
 	// Aslo, the first sample per each new series is ignored during stalenessSecs even if keepFirstSample is set.
-	// see ignoreFirstSampleDeadline for more details.
 	stalenessSecs uint64
-
-	// The first sample per each new series is ignored until this unix timestamp deadline in seconds even if keepFirstSample is set.
-	// This allows avoiding an initial spike of the output values at startup when new time series
-	// cannot be distinguished from already existing series. This is tracked with ignoreFirstSampleDeadline.
-	ignoreFirstSampleDeadline uint64
```

Comment:
I think this functionality isn't related to functionality of discarding "too old" samples. It is applied only to the first recorded sample. While `discard_samples_older_than` is applied to the whole data stream. Can we return it back?